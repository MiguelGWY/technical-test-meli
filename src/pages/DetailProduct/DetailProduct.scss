.container-detail {
  display: flex;
  flex-direction: column;
  align-items: center;

  &__breadcrumbs {
    width: 900px;
    height: 50px;
  }

  &__product {
    width: 900px;
    padding: 15px;
    background-color: white;
    border-radius: 5px;
  }
}
