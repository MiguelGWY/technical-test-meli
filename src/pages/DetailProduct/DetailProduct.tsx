import { useEffect, useState } from "react";
import { HeaderComponent } from "../../components/Header/HeaderComponent";
import { getItemById } from "../../api/services/productsById.service";
import { BreadCrumbs } from "../../components/BreadCrumbs/BreadCrumbsComponent";
import { getCategoriesById } from "../../api/services/categoriesById.service";
import { DetailProductCard } from "../../components/DetailProductCard/DetailProductCardComponent";
import { useLocation } from "react-router-dom";
import "./DetailProduct.scss";

export const DetailProduct = () => {
  const nameSpace = "container-detail";
  const [itemData, setItemData] = useState<ItemDetailResponse | null>(null);
  const location = useLocation();
  const idProduct = location.pathname.split("/").pop();

  useEffect(() => {
    const fetchProductData = async () => {
      try {
        if (idProduct) {
          const product = await getItemById(idProduct);
          const productCategories = await getCategoriesById(
            product.item.category_id
          );
          const itemDetailResponse: ItemDetailResponse = {
            author: product.author,
            item: {
              ...product.item,
              ...productCategories,
            },
          };
          setItemData(itemDetailResponse);
        }
      } catch (error) {
        console.error(
          "Se presentó un error en la petición del producto",
          error
        );
      }
    };

    fetchProductData();
  }, [idProduct]);

  return (
    <>
      <HeaderComponent />
      <section className={nameSpace}>
        <div className={`${nameSpace}__breadcrumbs`}>
          <BreadCrumbs categories={itemData?.item.categories || []} />
        </div>
        <div className={`${nameSpace}__product`}>
          {itemData && <DetailProductCard {...itemData.item} />}
        </div>
      </section>
    </>
  );
};
