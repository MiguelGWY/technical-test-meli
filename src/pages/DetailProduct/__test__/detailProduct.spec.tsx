import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import { DetailProduct } from "../DetailProduct";
import { MemoryRouter } from "react-router-dom";

describe("DetailProduct", () => {
  it("should render DetailProduct", () => {
    const { container } = render(
      <MemoryRouter>
        <DetailProduct />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
