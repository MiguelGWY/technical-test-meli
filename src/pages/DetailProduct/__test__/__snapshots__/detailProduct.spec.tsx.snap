// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailProduct should render DetailProduct 1`] = `
<div>
  <header
    class="nav-header"
    data-testid="header-component"
  >
    <div
      class="nav-header__container"
    >
      <div
        class="nav-header__logo"
      >
        <button
          aria-label="Regresar a la home"
          class="nav-header__redirect-button"
        >
          <img
            alt="logo mercadolibre"
            src="https://http2.mlstatic.com/frontend-assets/ml-web-navigation/ui-navigation/6.6.43/mercadolibre/logo_large_25years_v2.png"
          />
        </button>
      </div>
      <div
        class="nav-header__search"
      >
        <form
          class="nav-header__form"
          data-testid="form-input"
        >
          <label
            class="nav-header__label"
          >
            Ingresa lo que quieras encontrar
          </label>
          <input
            class="nav-header__search-input"
            placeholder="Nunca dejes de buscar"
            type="text"
            value=""
          />
        </form>
      </div>
    </div>
  </header>
  <section
    class="container-detail"
  >
    <div
      class="container-detail__breadcrumbs"
    />
    <div
      class="container-detail__product"
    />
  </section>
</div>
`;
